Classes/Indexer/NodeIndexer.php

```diff
@@ -339,19 +346,16 @@ protected function toBulkRequest(NodeInterface $node, array $requests = null): v
      * Returns a stable identifier for the Elasticsearch document representing the node
      *
      * @param NodeInterface $node
-     * @param string $targetWorkspaceName
+     * @param string|null $targetWorkspaceName
      * @return string
      * @throws IllegalObjectTypeException
      */
-    protected function calculateDocumentIdentifier(NodeInterface $node, $targetWorkspaceName = null): string
+    public static function calculateDocumentIdentifier(NodeInterface $node, $targetWorkspaceName = null): string
     {
-        $contextPath = $node->getContextPath();
-
-        if ($targetWorkspaceName !== null) {
-            $contextPath = (string)(new TargetContextPath($node, $targetWorkspaceName, $contextPath));
-        }
+        $workspaceName = $targetWorkspaceName ?: $node->getWorkspace()->getName();
+        $nodeIdentifier = $node->getIdentifier();
 
-        return sha1($contextPath);
+        return sha1($nodeIdentifier . $workspaceName);
```

Comment:
Currently, the contextPath contains, site, workspace, dimensions, and nodeIdentifier.
Dimensions should/may be covered, as the are going int separate indices. But sites are currently combined in the indices by dimension. Could that be an issue when cloning a site where somehow the nodeIdentfiers are the same. 

Why don't we just use the persistent_object_identifier, which is already unique?

Reply:
If you have two sites with nodes sharing the same identifier, you are doomed, I think. That should never happen, no? Then again, it could happen, so adding the site here might be good.

Using the `persistent_object_identifier` OTOH is a no-go IMHO. It should stay an implementation detail internal to the CR and not be used outide it.